app/partials/app.cjsx

```diff
@@ -3,15 +3,81 @@ auth = require 'panoptes-client/lib/auth'
 IOStatus = require './io-status'
 MainHeader = require './main-header'
 MainFooter = require './main-footer'
+GeordiClient = require 'zooniverse-geordi-client'
 {generateSessionID} = require '../lib/session'
 
+class GeordiLogger
+  constructor: (@state, @geordi) ->
+
+  keys: {}
+
+  geordi: null
+
+  instance: =>
+    @geordi = @geordi || @makeGeordi @keys?.projectToken
+
+  makeGeordi: (projectSlug) ->
+    new GeordiClient
+      server: @state?.env
+      projectToken: projectSlug || ''
```

Comment:
@alexbfree why do we need to instantiate a new Geordi client each time that the project slug changes? Shouldn't the client be instantiated once, then updated with a new project ID when the slug changes in the URL?

Other API clients (Panoptes, Talk, Sugar, Stats) are in `panoptes-client`, for convenience. Should the Geordi Panoptes client code live there too?